### PR TITLE
Clean up documentation and error handling around CKAN filter queries.

### PR DESCRIPTION
**Pull request by stevage**

### CODE REVIEW

File: lib/Models/CkanCatalogGroup.js [hunk 1]
```diff
@@ -305,6 +307,13 @@ CkanCatalogGroup.prototype._load = function() {
     var that = this;
 
     var promises = [];
+    if (!defined(this.filterQuery) || typeof this.filterQuery !== 'object' || typeof this.filterQuery[0] !== 'string') {
```
Comment (AlexGilleran) [inline lib/Models/CkanCatalogGroup.js]: You can use `Array.isArray()` in IE9+

**Comment by AlexGilleran** [comment]
Looks fine apart from the edge case where filterQuery is an object but not an array, not sure if that's possible?

Also is there any simple way to test it?
